flowmachine/flowmachine/core/server/query_schemas/aggregate_network_objects.py

```diff
@@ -0,0 +1,61 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marshmallow import Schema, fields, post_load
+from marshmallow.validate import OneOf, Length
+from marshmallow_oneofschema import OneOfSchema
+
+from flowmachine.features import AggregateNetworkObjects
+from .base_exposed_query import BaseExposedQuery
+from .total_network_objects import TotalNetworkObjectsSchema, TotalNetworkObjectsExposed
+from .custom_fields import AggregationUnit, Statistic, Period
+
+__all__ = ["AggregateNetworkObjectsSchema", "AggregateNetworkObjectsExposed"]
+
+
+class InputToAggregateNetworkObjectsSchema(OneOfSchema):
+    type_field = "query_kind"
+    type_schemas = {"total_network_objects": TotalNetworkObjectsSchema}
+
+
+class AggregateNetworkObjectsSchema(Schema):
+
+    total_network_objects = fields.Nested(
+        InputToAggregateNetworkObjectsSchema, required=True
+    )
+    statistic = Statistic()
+    period = Period()
```

Comment:
Ah no, right. I see, they're used here but not in TNO. I don't like the inconsistency, also the two arguments (`by` and `period`) have different allowable values. The `period` for TNO goes up to year, `by` for ANO goes to century. Looks like that isn't apparent from the docstring of the class, but is in fact the behaviour one wants.

Reply:
Yes I noticed the multiple "if" where the default ANO "period" is stepped up a notch relative to the TNO "by".